ql/src/java/org/apache/hadoop/hive/ql/optimizer/metainfo/annotation/OpTraitsRulesProcFactory.java

```diff
@@ -224,22 +273,34 @@ public Object process(Node nd, Stack<Node> stack, NodeProcessorCtx procCtx,
       } catch (HiveException e) {
         prunedPartList = null;
       }
-      boolean isBucketed = checkBucketedTable(table,
-          opTraitsCtx.getParseContext(), prunedPartList);
-      List<List<String>> bucketColsList = new ArrayList<List<String>>();
-      List<List<String>> sortedColsList = new ArrayList<List<String>>();
-      int numBuckets = -1;
-      if (isBucketed) {
+
+      final List<List<String>> bucketColsList = new ArrayList<>();
+      final List<List<String>> sortedColsList = new ArrayList<>();
+      final List<CustomBucketFunction> bucketFunctions = new ArrayList<>();
+      final int numBuckets;
```

Comment:
same here, why not set the default?

Reply:
Done
https://github.com/apache/hive/pull/5409/commits/c6aa9d2d0a81ff2e8834e1350a18405e593fe212